Support default network modes, Update Python and Django Support

rajadain opened the pull request:
## Overview

In the case of default network modes, the `networkMode` attribute is not specified in the `describe-task-definition` response. For these cases, we add tolerance for when it isn't specified, defaulting to `None`, and skipping the AWSVPC configuration block.

## BREAKING CHANGE

Also drops support for Django 2.2, 3.0, 3.1, and Python 3.6 and 3.7.

## Testing Instructions

- Check out this branch: https://github.com/azavea/collab-for-children/tree/release/1.0
- Run `./scripts/manage ecsmanage showmigrations`
  - [ ] Ensure it runs correctly

aaronxsu reviewed (commented):
I see that new commits are being pushed up, so unsure if it is related. I followed the testing steps on the collab for children side and ran the suggested command at the root of the project, and got the following. Lemme know if I missed anything.

```
 ~/Documents/collab-for-children -  (release/1.0) $ ./scripts/manage ecsmanage showmigrations
[+] Building 0.0s (0/0)                                                                                                                                                                        docker:desktop-linux
[+] Creating 1/0
 ✔ Container collab-for-children-database-1  Running                                                                                                                                                           0.0s 
[+] Building 0.0s (0/0)                                                                                                                                                                        docker:desktop-linux
Unknown command: 'ecsmanage'
Type 'manage.py help' for usage.
```

rajadain commented:
Ah right sorry, you need to run `update` after checking out that branch on `collab-for-children` to install the new dependency.

aaronxsu commented on .github/workflows/continuous_integration.yml:
```diff
@@ -13,12 +13,12 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: ["3.6", "3.7", "3.8"]
```
Will this affect Python versions used in projects?

rajadain replied:
This is what we're testing against, but we didn't change any of the code, so users running Python 3.6+ should be able to run this no problem, just as they are now.

rajadain committed:
Remove support for EOL Python, Django

rajadain committed:
Add support for new Python, Django

aaronxsu reviewed (approved):
Tested and works as advertised. Looks like there is a changelog file at the root of this project- might be good to append an entry there. Also, do we need to cut a release for this project?
```shell
~/Documents/collab-for-children -  (release/1.0) $ ./scripts/manage ecsmanage showmigrations
[+] Building 0.0s (0/0)                                                                                                                                                                        docker:desktop-linux
[+] Creating 1/0
 ✔ Container collab-for-children-database-1  Running                                                                                                                                                           0.0s 
[+] Building 0.0s (0/0)                                                                                                                                                                        docker:desktop-linux
Task started! View here:
https://console.aws.amazon.com/ecs/home?region=us-east-1#/clusters/ecsStagingCluster/tasks/aa0cc0bb39ff4a7fb4381be3c3c00036/details
```

rajadain committed:
Upgrade CI to use current GitHub Actions

Also upgrade environment variable passing for modern format

rajadain committed:
Support default network modes

In the case of EC2 launch type instances with no custom
network modes specified, the response of describe-task-definition
does not include a networkMode key. In those cases, we can
skip the AWSVPC block entirely. This adds tolerance for the
absence of that key.

rajadain commented:
Yes this will require a new release. Since we are dropping support for EOL'd Python and Django versions with this, I will do a major version upgrade. All existing projects should be unaffected since they are pinned to pull a current or older version, which does support their environment.

aaronxsu commented:
Awesome 🚀

rajadain commented:
Thanks for reviewing! And sorry about the in-flight commits 🙈 I didn't realize the build would break until I opened the PR.

aaronxsu commented:
No worries. All good.
